app/controllers/nonsipp/memberdetails/CheckMemberDetailsFileController.scala

```diff
@@ -63,26 +63,39 @@ class CheckMemberDetailsFileController @Inject()(
   private val form = CheckMemberDetailsFileController.form(formProvider)
 
   def onPageLoad(srn: Srn, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn).async { implicit request =>
-    val uploadStatus = request.userAnswers.get(UploadStatusPage(srn))
+    val uploadStatusOpt = request.userAnswers.get(UploadStatusPage(srn))
 
-    uploadStatus match {
+    uploadStatusOpt match {
       case Some(UploadSubmitted) =>
         val startTime = System.currentTimeMillis
         val uploadKey = UploadKey.fromRequest(srn)
         val preparedForm = request.userAnswers.fillForm(CheckMemberDetailsFilePage(srn), form)
+
         uploadService.getUploadStatus(uploadKey).map {
-          case None => Redirect(controllers.routes.JourneyRecoveryController.onPageLoad())
-          case Some(upload: UploadStatus.Success) =>
-            auditUpload(srn, upload, startTime)
-            Ok(view(preparedForm, viewModel(srn, Some(upload.name), mode)))
-          case Some(failure: UploadStatus.Failed) =>
-            auditUpload(srn, failure, startTime)
-            logger.warn("Upload failed")
-            Ok(view(preparedForm, viewModel(srn, Some(""), mode)))
-          case Some(_: UploadStatus.InProgress.type) =>
-            logger.info("Upload In progress, refreshing page")
-            Ok(view(preparedForm, viewModel(srn, None, mode)))
+          case None =>
+            Redirect(controllers.routes.JourneyRecoveryController.onPageLoad())
+
+          case Some(uploadStatus) =>
+            uploadStatus match {
+              case success: UploadStatus.Success =>
+                auditUpload(srn, success, startTime)
+                Ok(view(preparedForm, viewModel(srn, Some(success.name), mode)))
+
+              case failed: UploadStatus.Failed =>
+                auditUpload(srn, failed, startTime)
+                val errorForm = form.withError("file", "uploadMemberDetails.error.failed")
+                Ok(view(errorForm, viewModel(srn, None, mode)))
```

Comment:
@Arei1997 this will keep refreshing the page if the failure happens. One way to mitigate this would be to redirect to a different page and show an error there, as suggested on slack conversation. Might me a bit tricky how to force the error to be displayed properly.
Another way would be to extend viewModel with an additional parameter e.g. "doNotRefresh: Boolean = true" and make the refresh in viewModel conditional on that parameter too. May need to hide the radio on UI (or display a different view) and on button continue to `/upload-member-details-file`.